code/datums/materials/_material.dm

```diff
@@ -154,7 +154,11 @@ Simple datum which is instanced once per type and is used for every object of sa
 	// We assume no parallax means no space means no light
 	if(SSmapping.level_trait(on.z, ZTRAIT_NOPARALLAX))
 		return
-	on.set_light(2, 0.75, starlight_color)
+	on.set_light(2, 0.75, get_starlight_color())
+
+///Gets the space color and possible changed color if space is different
+/datum/material/proc/get_starlight_color()
+	return starlight_color || GLOB.starlight_color
```

Comment:
I can't tell off hand, do you use this proc anywhere else? I don't know if it's hot but adding such a small proc to something as common as materials, especially when this is being applied to an enormous amount of things with this trait gives me immense pause

Reply:
This only applies to transparent material turfs. On Roundstart, this is literally not called even once.

Glass tiles that are mapped in are handled somewhere else, but I did a similair thing there. They are called 26 times, because glass tiles just aren't used that much.

Actual space tiles are probably most costly, but it's just a single setter on Initialize()
![image](https://github.com/tgstation/tgstation/assets/7501474/76adb781-170e-402e-93d0-0ec79335f88c)